### PR TITLE
Implement Bundles

### LINKED ISSUE #604: Can't run Kingdom Hearts 1.5 + 2.5 HD ReMIX
Hello!
When i try to run the game (executable of the game or launcher executable it doesn't matter), i get a black screen and one error message spamming in the logs with infinite cycle:
`332:fixme:d3d12_command_list_update_current_framebuffer: Invalid RTV for attachment 0.`
And few other messages which appear very rarely. I'll attach the ghostbin with the log below.
https://ghostbin.com/paste/d2tlt

Will wait for the update on this. Thank You!

### PR DESCRIPTION
**Pull request by doitsujin**
Apparently fixes #604.

Edit: Marking as draft because there may be some ref counting weirdness left to test.

### CODE REVIEW

File: libs/vkd3d/bundle.c [hunk 1]
```diff
@@ -0,0 +1,1754 @@
...
+}
+
+static void d3d12_bundle_allocator_free_chunks(struct d3d12_bundle_allocator *allocator)
+{
+    size_t i;
+
+    for (i = 0; i < allocator->chunks_count; i++)
+        vkd3d_free(allocator->chunks[i]);
+
+    allocator->chunks_count = 0;
```
Comment (HansKristian-Work) [inline libs/vkd3d/bundle.c:66]: Shouldn't allocator->chunks also be freed?

File: libs/vkd3d/bundle.c [hunk 2]
```diff
@@ -0,0 +1,1754 @@
...
+        WARN("Bundle is not in the recording state.\n");
+        return E_FAIL;
+    }
+
+    bundle->is_recording = false;
+    return S_OK;
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_bundle_Reset(d3d12_command_list_iface *iface,
+        ID3D12CommandAllocator *allocator, ID3D12PipelineState *initial_pipeline_state)
```
Comment (HansKristian-Work) [inline libs/vkd3d/bundle.c:371]: initial_pipeline_state is not captured.

File: libs/vkd3d/device.c [hunk 3]
```diff
@@ -2613,39 +2613,33 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_CreateComputePipelineState(d3d12_d
...
+    if (FAILED(hr = ID3D12GraphicsCommandList_Reset(object, command_allocator, initial_pipeline_state)))
     {
-        WARN("Command list types do not match (allocator %#x, list %#x).\n",
-                allocator->type, type);
-        return E_INVALIDARG;
-    }
-
-    if (FAILED(hr = d3d12_command_list_create(device, node_mask, type, command_allocator,
-            initial_pipeline_state, &object)))
+        IUnknown_Release(object);
```
Comment (HansKristian-Work) [inline libs/vkd3d/device.c:2649]: Shouldn't this be ID3D12GraphicsCommandList_Release to avoid pointer warnings?

File: libs/vkd3d/bundle.c [hunk 4]
```diff
@@ -0,0 +1,1754 @@
...
+            sizeof(*args) + sizeof(*views) * view_count);
+    args->start_slot = start_slot;
+    args->view_count = view_count;
+    memcpy(args->views, views, sizeof(*views) * view_count);
+}
+
+static void STDMETHODCALLTYPE d3d12_bundle_SOSetTargets(d3d12_command_list_iface *iface,
+        UINT start_slot, UINT view_count, const D3D12_STREAM_OUTPUT_BUFFER_VIEW *views)
+{
+    WARN("iface %p, start_slot %u, view_count %u, views %p ignored!\n", iface, start_slot, view_count, views);
```
Comment (HansKristian-Work) [inline libs/vkd3d/bundle.c:1103]: Is this actually banned, or a FIXME?

File: libs/vkd3d/bundle.c [hunk 5]
```diff
@@ -0,0 +1,1754 @@
...
+    struct d3d12_bundle_command command;
+    FLOAT min;
+    FLOAT max;
+};
+
+static void d3d12_bundle_exec_om_set_depth_bounds(d3d12_command_list_iface *list, const void *args_v)
+{
+    const struct d3d12_om_set_depth_bounds_command *args = args_v;
+
+    ID3D12GraphicsCommandList1_OMSetDepthBounds(list, args->min, args->max);
```
Comment (HansKristian-Work) [inline libs/vkd3d/bundle.c:1323]: Pointer mismatch warnings?

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Mostly just nits. Looks fine overall. Not sure how many of the WARNs are actually supposed to not be supported, but I guess we'll just implement as needed since there are no good docs on the subject.

File: libs/vkd3d/bundle.c (hunk 4 above)
Comment (doitsujin) [reply libs/vkd3d/bundle.c:1103]: Banned.

File: libs/vkd3d/bundle.c (hunk 5 above)
Comment (doitsujin) [reply libs/vkd3d/bundle.c]: Should be fixed, but will require adjusting every single function call any time we implement a new version of the command list interface (C is a great language for this kind of thing...)

File: libs/vkd3d/device.c (hunk 3 above)
Comment (doitsujin) [reply libs/vkd3d/device.c]: Fixed.

File: libs/vkd3d/bundle.c (hunk 2 above)
Comment (doitsujin) [reply libs/vkd3d/bundle.c:376]: Fixed.

File: libs/vkd3d/bundle.c (hunk 1 above)
Comment (doitsujin) [reply libs/vkd3d/bundle.c:68]: The original idea was to recycle the array but right now we're not freeing it anywhere, so yeah, added that as well.

**Review by HansKristian-Work — APPROVED** [review]
LGTM!

**Pull request merged**